addons/spreadsheet/static/src/chart/plugins/odoo_chart_feature_plugin.js

```diff
@@ -0,0 +1,101 @@
+import { _t } from "@web/core/l10n/translation";
+import { OdooUIPlugin } from "@spreadsheet/plugins";
+import { getBestGranularity, getMinGranularity } from "../../global_filters/helpers";
+
+export class OdooChartFeaturePlugin extends OdooUIPlugin {
+    static getters = /** @type {const} */ (["getAvailableChartGranularities"]);
+    handle(cmd) {
+        switch (cmd.type) {
+            case "SET_GLOBAL_FILTER_VALUE": {
+                if (!this.getters.isDashboard()) {
+                    return;
+                }
+                const filterId = cmd.id;
+                const globalFilter = this.getters.getGlobalFilter(filterId);
+                if (globalFilter.type === "date") {
+                    const charts = this.getters.getOdooChartIds();
+                    for (const chartId of charts) {
+                        const { fieldName, granularity } =
+                            this.getters.getChartGranularity(chartId);
+                        const fieldMatching = this.getters.getChartFieldMatch(chartId)[filterId];
+                        const bestGranularity = getBestGranularity(cmd.value, fieldMatching);
+                        if (fieldMatching.chain === fieldName && bestGranularity !== granularity) {
+                            this.dispatch("UPDATE_CHART_GRANULARITY", {
+                                chartId,
+                                granularity: bestGranularity,
+                            });
+                        }
```

Comment:
Do we really want to arbitrary always change the granularity ? What if the chart is originally grouped by week, do we really want to change it to month when updating the global filters ?